docs-src/doctype-starter-doc.md

```diff
@@ -0,0 +1,362 @@
+# Create a Ceramic Doctype
+
+In order to implement a new Ceramic Doctype there are a couple of interfaces which should be followed:
+
+- **Doctype**
+	- This interface is used to implement an actual document.
+- **DoctypeHandler**
+	- This interface is used to implement the state transition handler for the document which will be executed by the Ceramic node.
+
+There are other constructs which need to be used in order to create a `Doctype`:
+- **DoctypeStatic**
+	- Decorator for the Doctype implementation
+- **DoctypeConstructor**
+	- Describes the Doctype constructor which needs to be implemented.
+
+One interface worth noting is the `CeramicAPI` interface which lets the developer communicate with the Ceramic node in two ways:
+- **directly**: the node is the process running on the same machine
+- **indirectly**: the node is running somewhere else and the communication is done over the HTTP protocol.
+	
+# Doctype
+The interface is listed in the code snipped below.
+
+```typescript
+/**  
+ * Describes common doctype attributes 
```

Comment:
Instead of copying code and embedding it into the docs where it can quickly get out of sync, should we just be linking to the relevant code in github, perhaps pinned to the last stable release version?

Reply:
Good point, mainly copied this document over. I think we can now actually link the correct code in the docsite